### PR TITLE
Accept JSON as payload for relay 's /v1/callbacks

### LINKED ISSUE #939: Bonsai 's relay REST /v1/callbacks accepts only bincode serialized data
This limits the possibility of crafting CallBackRequests in other languages and limits it to rust. 

```rs
/// Post a new [CallbackRequest] to Bonsai.
    pub async fn callback_request(
        &self,
        request: impl Into<CallbackRequest>,
    ) -> Result<(), ClientError> {
        let res = self
            .client
            .post(format!("{}{CALLBACK_ROUTE}", self.url))
            .header(header::CONTENT_TYPE, "application/octet-stream")
            .body(bincode::serialize(&request.into())?)
            .send()
            .await?;
        error_for_status(res).await?;

        Ok(())
    }
```

A more widely accepted format would be JSON bodies or BASE64 parameters

### PR DESCRIPTION
**Pull request by DefiCake**
Closes #939

### CODE REVIEW

**Comment by mothran** [comment]
Approved for CI and added some reviewers who can help you out.

File: bonsai/ethereum-relay/src/api/bincode.rs [hunk 1]
```diff
@@ -30,39 +30,62 @@ pub(crate) struct Bincode<T>(pub T);
...
+            })?;
 
-        Ok(Bincode(value))
+            return Ok(Bincode(value));
+        }
+
+        Err(
+            (
+                StatusCode::UNSUPPORTED_MEDIA_TYPE,
+                "Expected request with `Content-Type: application/octet-stream`",
```
Comment (capossele) [inline bonsai/ethereum-relay/src/api/bincode.rs:86]:
```suggestion
                "Expected request with `Content-Type: application/octet-stream` or `Content-Type: application/json`",
```

File: bonsai/ethereum-relay/src/api/bincode.rs [hunk 2]
```diff
@@ -30,39 +30,62 @@ pub(crate) struct Bincode<T>(pub T);
...
-    B: HttpBody + Send + 'static,
-    B::Data: Send,
-    B::Error: std::error::Error + Send + Sync,
-    S: Send + Sync,
+    where
+        T: DeserializeOwned,
+        B: HttpBody + Send + 'static,
+        B::Data: Send,
+        B::Error: std::error::Error + Send + Sync,
+        S: Send + Sync,
```
Comment (capossele) [inline bonsai/ethereum-relay/src/api/bincode.rs:38]:
```suggestion
where
    T: DeserializeOwned,
    B: HttpBody + Send + 'static,
    B::Data: Send,
    B::Error: std::error::Error + Send + Sync,
    S: Send + Sync,
```

**Review by capossele — CHANGES_REQUESTED** [review]
Thanks for that!
Overall it's a nice addition. I have one observation:
The `Bincode` extractor is now mixed with json. I'd suggest to rename it to something more neutral that works for json as well. Same goes for the its documentation.
Another plus would be to have a test in place that make sure both bincode and json encoding deserialise correctly to a `CallbackRequest`.

**Review by nategraf — COMMENTED** [review]
Looks good! Similar to @caposselemy only comment is to rename the file and type to reflect the new more general usage 👍 Thanks for getting involved!

**Comment by DefiCake** [comment]
I think the CI is failing due to the last commit in main branch

**Comment by capossele** [comment]
but yeah, the rest of the tests are failing for some other reason. We'll sort that out

**Comment by DefiCake** [comment]
@capossele fmt run, I 'll await to update wit hthe main branch

**Review by capossele — APPROVED** [review]
Awesome, thanks!

**Pull request merged**